fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -4014,24 +4015,42 @@ public void alterTableProperties(Database db, OlapTable table, Map<String, Strin
+            if (partitionLiveNumber == TableProperty.INVALID) {
+                GlobalStateMgr.getCurrentState().getDynamicPartitionScheduler().removeTtlPartitionTable(db.getId(),
+                        table.getId());
+            } else {
+                GlobalStateMgr.getCurrentState().getDynamicPartitionScheduler().registerTtlPartitionTable(db.getId(),
+                        table.getId());
+            }
             tableProperty.setPartitionTTLNumber(partitionLiveNumber);
-        } else if (logProperties.containsKey(PropertyAnalyzer.PROPERTIES_STORAGE_MEDIUM)) {
+            ModifyTablePropertyOperationLog info =
+                    new ModifyTablePropertyOperationLog(db.getId(), table.getId(),
+                            ImmutableMap.of(PropertyAnalyzer.PROPERTIES_PARTITION_LIVE_NUMBER,
+                                    logProperties.get(PropertyAnalyzer.PROPERTIES_PARTITION_LIVE_NUMBER)));
+            GlobalStateMgr.getCurrentState().getEditLog().logAlterTableProperties(info);
+        }
+        if (logProperties.containsKey(PropertyAnalyzer.PROPERTIES_STORAGE_MEDIUM)) {
             TStorageMedium storageMedium = dataProperty.getStorageMedium();
             table.setStorageMedium(storageMedium);
             tableProperty.getProperties()
                     .put(PropertyAnalyzer.PROPERTIES_STORAGE_COOLDOWN_TIME,
                             String.valueOf(dataProperty.getCooldownTimeMs()));
-        } else if (logProperties.containsKey(PropertyAnalyzer.PROPERTIES_STORAGE_COOLDOWN_TTL)) {
+            ModifyTablePropertyOperationLog info =
+                    new ModifyTablePropertyOperationLog(db.getId(), table.getId(),
+                            ImmutableMap.of(PropertyAnalyzer.PROPERTIES_STORAGE_MEDIUM,
+                                    logProperties.get(PropertyAnalyzer.PROPERTIES_STORAGE_MEDIUM)));
+            GlobalStateMgr.getCurrentState().getEditLog().logAlterTableProperties(info);
+        }
+        if (logProperties.containsKey(PropertyAnalyzer.PROPERTIES_STORAGE_COOLDOWN_TTL)) {
             String storageCoolDownTTL = logProperties.get(PropertyAnalyzer.PROPERTIES_STORAGE_COOLDOWN_TTL);
             tableProperty.getProperties().put(PropertyAnalyzer.PROPERTIES_STORAGE_COOLDOWN_TTL, storageCoolDownTTL);
             tableProperty.buildStorageCoolDownTTL();
-        } else {
-            throw new DdlException("Modify failed because unknown properties: " + properties);
+            ModifyTablePropertyOperationLog info =
+                    new ModifyTablePropertyOperationLog(db.getId(), table.getId(),
+                            ImmutableMap.of(PropertyAnalyzer.PROPERTIES_STORAGE_COOLDOWN_TTL,
+                                    logProperties.get(PropertyAnalyzer.PROPERTIES_STORAGE_COOLDOWN_TTL)));
+            GlobalStateMgr.getCurrentState().getEditLog().logAlterTableProperties(info);
         }
-
-        ModifyTablePropertyOperationLog info =
```

Comment:
Why write multi edit log instead of only one?

Reply:
Because this can only support one log and one process.